enhancements/microshift/microshift-router-configuration-performance.md

```diff
@@ -0,0 +1,196 @@
+shortening the timeout, allowing connections to be closed more aggressively.
+
+
+### User Stories
+TBD
+
+### Goals
+Allow users to configure the following HAProxy connection timeout parameters:
+- timeout client
+- timeout client-fin
+- timeout server
+- timeout server-fin
+- timeout tunnel
+- tcp-request inspect-delay
+
+### Non-Goals
+N/A
+
+## Proposal
+
+see the table below for the proposed changes:
+
+| new configuration | description | New var? | haproxy ENV Vars [src](https://github.com/openshift/router/blob/master/images/router/haproxy/conf/haproxy-config.template) | default | Type |
+|---|---|---|---|---|---|
+| MimeTypes |  | Yes | <ul> <li>ROUTER_COMPRESSION_MIME</li> <li>ROUTER_ENABLE_COMPRESSION</li> </ul> | none |  |
+| httpHeaders | defines the policy for HTTP headers, options: <ul> <li>forwardedHeaderPolicy</li> <li>forwardedHeaderPolicys</li> </ul> | No | ROUTER_SET_FORWARDED_HEADERS | append | Enum: Append,Replace,IfNone,Never |
+| tuningOptions -> headerBufferBytes |  | Yes | ROUTER_BUF_SIZE | 32768 |  |
+| tuningOptions -> headerBufferMaxRewriteBytes |  | Yes | ROUTER_MAX_REWRITE_SIZE | 8192 |  |
+| tuningOptions -> healthCheckInterval |  | Yes | ROUTER_BACKEND_CHECK_INTERVAL | 5000ms |  |
+| tuningOptions -> clientTimeout |  | Yes | ROUTER_DEFAULT_CLIENT_TIMEOUT | 30s |  |
+| tuningOptions -> clientFinTimeout |  | Yes | ROUTER_CLIENT_FIN_TIMEOUT | 1s |  |
+| tuningOptions -> serverTimeout |  | Yes | ROUTER_DEFAULT_SERVER_TIMEOUT | 30s |  |
+| tuningOptions -> serverFinTimeout |  | Yes | ROUTER_DEFAULT_SERVER_FIN_TIMEOUT | 1s |  |
+| tuningOptions -> tunnelTimeout |  | Yes | ROUTER_DEFAULT_TUNNEL_TIMEOUT | 1h |  |
+| tuningOptions -> tlsInspectDelay |  | Yes | ROUTER_INSPECT_DELAY | 5s |  |
+| tuningOptions -> threadCount |  | No | ROUTER_THREADS | 4 |  |
+| tuningOptions -> maxConnections |  | Yes | ROUTER_MAX_CONNECTIONS | 50000 |  |
+| LogEmptyRequests |  | Yes | ROUTER_DONT_LOG_NULL |  |  |
+| HTTPEmptyRequestsPolicy |  | Yes | ROUTER_HTTP_IGNORE_PROBES |  |  |
+| HTTP2 |  | NO | ROUTER_DISABLE_HTTP2 | true | Bool |
+
+### Configure IPs where the router listens
+The following configuration is proposed:
+```yaml
+ingress:
+    - ...
+```
+
+### Workflow Description
+**cluster admin** is a human user responsible for configuring a MicroShift
+cluster.
+
+1. The cluster admin adds specific configuration for the router prior to
+   MicroShift's start.
+2. After MicroShift started, the system will ingest the configuration and setup
+   everything according to it.
+
+
+### API Extensions
+As described in the proposal, there is an entire new section in the configuration:
+```yaml
+ingress:
+    - ...
+```
+
+For more information check each individual section.
+
+### Topology Considerations
+TBD
+
+#### Standalone Clusters
+N/A
+
+#### Single-node Deployments or MicroShift
+Enhancement is solely intended for MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+
+
+### Risks and Mitigations
+TBD
+
+### Drawbacks
```

Comment:
So what are the in-the-field consequences of either scenario? For example, brick the device in #1? Cause failure of task in scenario #2? Do we want to give the users any quantifiable guidance about values to set?

Reply:
well can we warn  customer that the defaults shouldn't be changed unless he is really knows what he is doing..

Reply:
its really the internals of the routers